engine/runtime/src/main/java/org/enso/interpreter/node/callable/IndirectInvokeMethodNode.java

```diff
@@ -129,9 +131,13 @@ Object doWarning(
       InvokeCallableNode.ArgumentsExecutionMode argumentsExecutionMode,
       BaseNode.TailStatus isTail,
       int thisArgumentPosition,
-      @Cached IndirectInvokeMethodNode childDispatch) {
+      @Cached IndirectInvokeMethodNode childDispatch,
+      @Cached AppendWarningNode appendWarningNode,
+      @CachedLibrary(limit = "3") WarningsLibrary warnsLib,
+      @Cached HashMapInsertAllNode mapInsertAllNode,
+      @CachedLibrary(limit = "3") InteropLibrary interop) {
     arguments[thisArgumentPosition] = self.getValue();
-    ArrayRope<Warning> warnings = self.getReassignedWarningsAsRope(this, false);
+    Warning[] warnings = self.getWarningsArray(false, warnsLib, mapInsertAllNode, interop);
```

Comment:
Another place for `EnsoHashMap` I think.

Reply:
Right. Migrated to `EnsoHashMap` in a88ae7489b6778b02ac5e66266620a982b53b113